src/cast_core/test/AssertiveActor_test.cpp

```diff
@@ -0,0 +1,218 @@
+            SchemaVersion: 2018-07-01
+            Clients:
+              Default:
+                URI: )" + MongoTestFixture::connectionUri().to_string() + R"(
+            Actors:
+            - Name: )" + actorName + R"(
+              Type: AssertiveActor
+              Database: test
+              Phases:
+              - Repeat: 1
+                Expected:
+                  aggregate: expected
+                  pipeline: []
+                  cursor: {batchSize: 101}
+                Actual:
+                  aggregate: actual
+                  pipeline: []
+                  cursor: {batchSize: 101}
+                )" + ignoreFieldsStr;
+        NodeSource nodes = NodeSource(yaml, __FILE__);
+        return genny::ActorHelper(nodes.root(), 1);
+    }
+
+    static void expectAssertPasses(
+        const std::string& actorName,
+        const std::string& ignoreFields = ""
+    ) {
+        try {
+            genny::ActorHelper ah = setupAssertActor(actorName, ignoreFields);
+            ah.run([](const genny::WorkloadContext& wc) { wc.actors()[0]->run(); });
+            INFO("Assert passed")
+        } catch (const genny::actor::FailedAssertionException& e) {
+            // Did not expect assert to reach here.
+            auto diagInfo = boost::diagnostic_information(e);
+            FAIL("Assert failed" << diagInfo);
+        }
+    }
+
+    static void expectAssertFails(
+        const std::string& actorName,
+        const std::string& ignoreFields = ""
+    ) {
+        try {
+            genny::ActorHelper ah = setupAssertActor(actorName, ignoreFields);
+            ah.run([](const genny::WorkloadContext& wc) { wc.actors()[0]->run(); });
+            FAIL("Assert passed");
+        } catch (const genny::actor::FailedAssertionException& e) {
+            // Expected assert to reach here.
+            INFO("Assert failed");
+        }
+    }
+};
+
+TEST_CASE_METHOD(AssertiveActorTestFixture, "AssertiveActor passes an assert", "[standalone][single_node_replset][three_node_replset][sharded][AssertiveActor]") {
+    prepareDatabase();
+  
+    // The test collections are empty, so this should trivially pass.
+    SECTION("Assert passes because empty collections are equivalent") {
+        AssertiveActorTestFixture::expectAssertPasses("EmptyCollections");
+    }
+
+    auto doc1 = bson_builder::make_document(
+        bson_builder::kvp("a", 1),
+        bson_builder::kvp("b", "foo"),
+        bson_builder::kvp("c", bson_builder::make_document(bson_builder::kvp("d", 1))),
+        bson_builder::kvp("d", bson_builder::make_array(1, 2, 3))
+    );
+    auto doc2 = bson_builder::make_document(
+        bson_builder::kvp("a", 1),
+        bson_builder::kvp("e", 1.4)
+    );
+    std::vector<bsoncxx::document::view> docs {doc1, doc2};
+    prepareDatabase(docs, docs);
+    SECTION("Assert passes because collections are equivalent") {
+        AssertiveActorTestFixture::expectAssertPasses("EqualCollections");
+    }
```

Comment:
Could you also add a few test cases to cover objects and arrays?

Reply:
Oh sorry I didn't see there are arrays in the "passing" block. nvm. Resolving this